p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -461,39 +461,64 @@ public boolean refreshTTL(RefreshOfferMessage refreshTTLMessage,
                               @Nullable NodeAddress sender,
                               boolean isDataOwner) {
         ByteArray hashOfPayload = new ByteArray(refreshTTLMessage.getHashOfPayload());
-        if (map.containsKey(hashOfPayload)) {
-            ProtectedStorageEntry storedData = map.get(hashOfPayload);
-            int sequenceNumber = refreshTTLMessage.getSequenceNumber();
 
-            if (sequenceNumberMap.containsKey(hashOfPayload) && sequenceNumberMap.get(hashOfPayload).sequenceNr == sequenceNumber) {
-                log.trace("We got that message with that seq nr already from another peer. We ignore that message.");
-                return true;
-            } else {
-                PublicKey ownerPubKey = storedData.getProtectedStoragePayload().getOwnerPubKey();
-                byte[] hashOfDataAndSeqNr = refreshTTLMessage.getHashOfDataAndSeqNr();
-                byte[] signature = refreshTTLMessage.getSignature();
-                // printData("before refreshTTL");
-                if (hasSequenceNrIncreased(sequenceNumber, hashOfPayload) &&
-                        checkIfStoredDataPubKeyMatchesNewDataPubKey(ownerPubKey, hashOfPayload) &&
-                        checkSignature(ownerPubKey, hashOfDataAndSeqNr, signature)) {
-                    log.debug("refreshDate called for storedData:\n\t" + StringUtils.abbreviate(storedData.toString(), 100));
-                    storedData.refreshTTL();
-                    storedData.updateSequenceNumber(sequenceNumber);
-                    storedData.updateSignature(signature);
-                    printData("after refreshTTL");
-                    sequenceNumberMap.put(hashOfPayload, new MapValue(sequenceNumber, System.currentTimeMillis()));
-                    sequenceNumberMapStorage.queueUpForSave(SequenceNumberMap.clone(sequenceNumberMap), 1000);
-
-                    broadcast(refreshTTLMessage, sender, null, isDataOwner);
-                    return true;
-                }
-
-                return false;
-            }
-        } else {
+        if (!map.containsKey((hashOfPayload))) {
             log.debug("We don't have data for that refresh message in our map. That is expected if we missed the data publishing.");
+
             return false;
         }
+
+        int sequenceNumber = refreshTTLMessage.getSequenceNumber();
+
+        // If we have seen a more recent operation for this payload, we ignore the current one
+        // TODO: I think we can return false here. All callers use the get() public API which increments the sequence number
+        // leaving only the onMessage() handler which doesn't look at the return value. It makes more intuitive sense that operations that don't
+        // change state return false.
+        if (sequenceNumberMap.containsKey(hashOfPayload) && sequenceNumberMap.get(hashOfPayload).sequenceNr == sequenceNumber) {
```

Comment:
In OfferBookService it is used and would cause a error message in that case (thought I think its an impossible case). Have not followed further if that error message would cause more troubles but I think it would require more checks of all client usage code to be sure we can change return value.  Conceptually I agree.